public/js/scripts.js

```diff
@@ -99,7 +99,7 @@ var defaultPostSize = 0;                //Default height of a post, used to dete
 var deviceType = "desktop";             //Initially assumes mobile and is changed further down if the user is on desktop
 var defaultLimit = 15;                  //default limit on number of questions to display
 var openMenu = "none";                  //currently opened menu (ex. 'sort', 'search', etc...) used in close menu function
-var acceptableLinkCharacters = "ABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890abcdefghijklmnopqrstuwxyz.:/?=_";  //characters that are allowed in a video link
+var acceptableLinkCharacters = "ABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890abcdefghijklmnopqrstuvwxyz.:/?=_";  //characters that are allowed in a video link
```

Comment:
I don't like the way the youtube parser looks for acceptable characters. Basically it continues to check every character in the reply until it finds one that is not on this list, and decides that location is the end of the youtube link. The only reason I did it this way is because this way works no mater where the link is in the reply, at the start, at the end of a line, or at the end or the reply with no characters after word it still finds the end. 

If someone could think of a better way to do this that would be great, because with the way it currently works, it breaks if you forget to put a character (like 'v') in the list (like I did). 
It works for now and is good enough for production.